test/MUXControlsTestApp/VisualTreeTestBase.cs

```diff
@@ -0,0 +1,360 @@
+            ExpectedMasterFileName = GetExpectedMasterFileName(prefix);
+            BestMatchedMasterFileName = SearchBestMatchedMasterFileName(prefix);
+            StorageLocation = useLocalStorage ? ApplicationData.Current.LocalFolder.Path : "MusicLibrary";
+        }
+
+        public void LogMasterFile(string fileName, string content)
+        {
+            LogMasterFile(_storage, fileName, content);
+        }
+
+        private string GetExpectedMasterFileName(string fileNamePrefix)
+        {
+            return String.Format("{0}-{1}.xml", fileNamePrefix, PlatformConfiguration.GetCurrentAPIVersion());
+        }
+
+        private string SearchBestMatchedMasterFileName(string fileNamePrefix)
+        {
+            for (ushort version = PlatformConfiguration.GetCurrentAPIVersion(); version >= 2; version--)
+            {
+                string fileName = String.Format("{0}-{1}.xml", fileNamePrefix, version);
+                if (MasterFileStorage.IsMasterFilePresent(fileName))
+                {
+                    return fileName;
+                }
+            }
+            {
+                string fileName = fileNamePrefix + ".xml";
+                if (MasterFileStorage.IsMasterFilePresent(fileName))
+                {
+                    return fileName;
+                }
+            }
+            return null;
+        }
+
+        public static string GetMasterFileContent(string fileName)
+        {
+            return GetMasterFileContentAsync(fileName).Result;
+        }
+
+        public static bool IsMasterFilePresent(string fileName)
+        {
+            try
+            {
+                GetMasterFileContentAsync(fileName).Wait();
+                return true;
+            }
+            catch (AggregateException ae)
+            {
+                ae.Handle((x) =>
+                {
+                    return (x is FileNotFoundException); // This we know how to handle.
+                });
+                return false;
+            }
+        }
+
+        public static void LogMasterFile(StorageFolder storageFolder, string fileName, string content)
+        {
+            LogMasterFileAsync(storageFolder, fileName, content).Wait();
+        }
+
+        private static readonly string MasterFileFullPathPrefix = "ms-appx:///master/";
+
+        private static async Task<string> GetMasterFileContentAsync(string fileName)
+        {
+            Uri uri = new Uri(MasterFileFullPathPrefix + fileName);
+            var file = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            return await FileIO.ReadTextAsync(file);
+        }
+
+        private static async Task LogMasterFileAsync(StorageFolder storageFolder, string fileName, string content)
+        {
+            var file = await storageFolder.CreateFileAsync(fileName, CreationCollisionOption.ReplaceExisting);
+            await FileIO.WriteTextAsync(file, content);
+        }
+    }
+
+    class VisualTreeOutputCompare
```

Comment:
Is there going to be a way to caputure the UI tree and then query for certain properties specifically ? What if i only care about some properties and i do not want to manage a baseline file.

Reply:
yes. Just call VisualTreeTestHelper.DumpVisualTree, and pass IFilter subclass as parameter.